tests/transforms/test_suite2p_wrapper.py

```diff
@@ -0,0 +1,137 @@
+import pytest
+from unittest.mock import MagicMock, Mock
+from pathlib import Path
+from functools import partial
+import os
+import numpy as np
+import h5py
+import json
+import filecmp
+import random
+
+import sys
+sys.modules['suite2p'] = Mock()
+from ophys_etl.transforms import suite2p_wrapper # noqa
+
+suite2p_basenames = ['ops1.npy', 'data.bin', 'Fneu.npy', 'F.npy', 'iscell.npy',
+                     'ops.npy', 'spks.npy', 'stat.npy']
+
+
+@pytest.mark.parametrize(
+        "basenames, dstsubdir, exception",
+        [
+            (['tmp.txt'], "other", False),
+            (['tmp.txt', "tmp2.txt"], "other", False),
+            (['tmp.txt', "tmp.txt"], "other", True),
+            ])
+def test_copy_and_add_uid(tmp_path, basenames, dstsubdir, exception):
+    srcfiles = {}
+    for i, bname in enumerate(basenames):
+        if i == 0:
+            tfile = tmp_path / bname
+        else:
+            spath = tmp_path / "sub_dir"
+            spath.mkdir(exist_ok=True)
+            tfile = spath / bname
+        with open(tfile, "w") as f:
+            f.write("%032x" % random.getrandbits(128))
+        srcfiles[bname] = str(tfile)
+
+    other_dir = tmp_path / "other"
+    other_dir.mkdir()
+
+    uid = "extra"
+    if exception:
+        with pytest.raises(ValueError):
+            dstfiles = suite2p_wrapper.copy_and_add_uid(
```

Comment:
I would test for the error you expect to be raised, pytest.raises has a match parameter that takes a regex that's helpful for this.

Reply:
done